scheduler/src/cook/kubernetes/controller.clj

```diff
@@ -140,6 +143,25 @@
     (write-status-to-datomic kcc status)
     {:expected-state :expected/running}))
 
+(defn record-sandbox-url
+  "Record the sandbox file server URL in datomic."
+  [{:keys [pod]}]
+  (let [task-id (-> pod .getMetadata .getName)
+        pod-ip (-> pod .getStatus .getPodIP)
+        {:keys [default-workdir sandbox-fileserver pod-ip->hostname-fn]} (config/kubernetes)
+        sandbox-fileserver-port (:port sandbox-fileserver)
+        sandbox-url (try
+                      (when (and sandbox-fileserver-port (not (str/blank? pod-ip)))
+                        (str "http://"
+                             (pod-ip->hostname-fn pod-ip)
+                             ":" sandbox-fileserver-port
+                             "/files/read.json?path="
+                             (URLEncoder/encode default-workdir "UTF-8")))
+                      (catch Exception e
+                        (log/debug e "Unable to retrieve directory path for" task-id)
+                        nil))]
+    (scheduler/write-sandbox-url-to-datomic datomic/conn task-id sandbox-url)))
```

Comment:
Do we actually need to write to datomic when the result was `nil`?

Reply:
can use the YAML directly to spin up the same pod
When will YAML be an issue? it's fine in elastic

there's another task for the other refactoring

Reply:
will not write nil